### ISSUE TITLE
Feat: <md-aria-status> directive to be consumed by mdToast and mdAutocomplete 

### ISSUE THREAD

**Issue by marcysutton** [angular/material#3174]
To enable `$mdToast` accessibility, Angular Material needs a global version of the `<aria-status>` directive used in `mdAutocomplete`. A single Live Region directive for an entire app, which could be used by multiple components as a reusable message center. It would also make screen reader messages easier to localize, per #1696. This should be part of the core functionality to ensure its inclusion for Angular Material components, though I could see a message service being useful for other live regions.

mdAutocomplete uses `aria-live="assertive"` so that messages will be announced immediately as you type, but I'm investigating whether it can use `aria-live="polite"`, which would be more appropriate for mdToast. Components using the global directive will have to share live region settings, as far as I know (role, atomic, relevant, etc.).

**Comment by marcysutton** [comment]
The `announce` feature of this library could be useful: https://github.com/dequelabs/ngA11y/blob/master/src/nga11yannounce.js

**Comment by naomiblack** [comment]
@jelbourn it would be really nice to do this in 1.0 bc of breaking change (and overall a11y), but I'm concerned about time needed to get it right. Do you think this is something you could squeak in along with other m12 tasks on your plate?

**Comment by jelbourn** [comment]
@marcysutton Could you elaborate on the problems with having multiple live regions on a page?

**Comment by marcysutton** [comment]
@jelbourn if there are multiple live regions doing the same thing on a page (and with the same politeness level), they can cause confusion in a screen reader–especially if we can't control how many components are used that require announcements. It makes sense for Angular Material to handle messages centrally to reduce complexity under the hood and optimize for assistive technology. Here is a little more info and a solution that seems to work but may need some more testing in IE: http://spa-a11y.divshot.io/#14
